### PR TITLE
MAGECLOUD-5306: Run the update command after enabling SPLIT_DB

### PR DESCRIPTION
**Pull request by arhiopterecs**
https://magento2.atlassian.net/browse/MAGECLOUD-5306

### Description
Adds the launch of a CLI command `bin\magento setup:upgrade` after enabling split databases

### Fixed Issues (if relevant)
https://magento2.atlassian.net/browse/MAGECLOUD-5306

### Manual testing scenarios
1. Enable split database in Magento Cloud for any type
    Expected: 
    The splitting went successful
2. Open '<magento_root>/var/log/cloud.log' log file
     Expected: 
     After executing each command for splitting databases, the upgrade command `php bin/magento setup:upgrade ` is executed 

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: src/Step/Deploy/InstallUpdate/Update/Setup.php [hunk 1]
```diff
@@ -7,109 +7,54 @@
...
-            $this->shell->execute(sprintf(
-                '/bin/bash -c "set -o pipefail; %s | tee -a %s"',
-                'php ./bin/magento setup:upgrade --keep-generated --ansi --no-interaction ' . $verbosityLevel,
-                $installUpgradeLog
-            ));
-        } catch (\RuntimeException $exception) {
+            $this->upgradeRunner->execute();
+        } catch (\Exception $exception) {
             //Rollback required by database
             throw new StepException($exception->getMessage(), 6, $exception);
```
Comment (oshmyheliuk) [inline src/Step/Deploy/InstallUpdate/Update/Setup.php]: What does it mean error code 6? If you are using custom codes is better to create constant with an appropriate name.

File: src/Util/UpgradeProcess.php [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+/**
+ * Utility class for run upgrade process.
+ */
+class UpgradeProcess
+{
+    /**
+     * @var ShellInterface
+     */
+    private $shell;
+    /**
```
Comment (oshmyheliuk) [inline src/Util/UpgradeProcess.php:28]: add empty line

File: src/Test/Unit/Config/SchemaTest.php [hunk 3]
```diff
@@ -117,6 +117,8 @@ public function testGetDefaultsForPostDeploy(): void
                     '',
                 ],
                 PostDeployInterface::VAR_TTFB_TESTED_PAGES => [],
+                PostDeployInterface::VAR_VERBOSE_COMMANDS => ''
+
```
Comment (bbatsche) [inline src/Test/Unit/Config/SchemaTest.php]: Remove extra blank line

**Review by bbatsche — CHANGES_REQUESTED** [review]

File: src/Step/Deploy/InstallUpdate/Update/Setup.php (hunk 1 above)
Comment (arhiopterecs) [reply src/Step/Deploy/InstallUpdate/Update/Setup.php]: @oshmyheliuk 
Thank you very much for the comment.
But could you explain the meaning of this value?
https://github.com/magento/ece-tools/commit/414899313c8cdcc5a27db525e4b8800ebe43b5a1#diff-1c1a326236ff13e56dca5f0910661aa1R83

File: src/Test/Unit/Config/SchemaTest.php (hunk 3 above)
Comment (arhiopterecs) [reply src/Test/Unit/Config/SchemaTest.php]: Done

File: src/Util/UpgradeProcess.php (hunk 2 above)
Comment (arhiopterecs) [reply src/Util/UpgradeProcess.php:28]: Done

File: src/Step/Deploy/InstallUpdate/Update/Setup.php (hunk 1 above)
Comment (arhiopterecs) [reply src/Step/Deploy/InstallUpdate/Update/Setup.php]: Done

Comment (oshmyheliuk) [reply src/Step/Deploy/InstallUpdate/Update/Setup.php]: Have no idea, I think it was ported from old MCC.

**Review by bbatsche — DISMISSED** [review]

File: src/Step/Deploy/SplitDbConnection.php [hunk 4]
```diff
@@ -64,43 +60,48 @@ class SplitDbConnection implements StepInterface
...
         DeployInterface $stageConfig,
         DbConfig $dbConfig,
-        ResourceConfig $resourceConfig,
         LoggerInterface $logger,
         FlagManager $flagManager,
         ConfigReader $configReader,
-        ConfigWriter $configWriter,
-        MagentoShell $magentoShell
+        MagentoShell $magentoShell,
+        UpgradeProcess $upgradeRunner,
```
Comment (oshmyheliuk) [inline src/Step/Deploy/SplitDbConnection.php]: Why did you name the class as `UpgradeProcess` but the variable name `$upgradeRunner`.
It is better to rename class to `UpgradeRunner` or variables to `$upgradeProcess` to make naming consistent.

**Review by billygilbert — DISMISSED** [review]

File: src/Step/Deploy/SplitDbConnection.php (hunk 4 above)
Comment (arhiopterecs) [reply src/Step/Deploy/SplitDbConnection.php]: My mistake, it was an old class name

Comment (arhiopterecs) [reply src/Step/Deploy/SplitDbConnection.php]: Done

**Review by oshmyheliuk — DISMISSED** [review]

**Review by bbatsche — DISMISSED** [review]

**Review by billygilbert — APPROVED** [review]

**Comment by andriyShevtsov** [comment]
QA approved

**Review by oshmyheliuk — APPROVED** [review]

**Pull request merged**